spec/tapioca/dsl/compilers/active_record_associations_spec.rb

```diff
@@ -1603,6 +1603,249 @@ def shop_previously_changed?; end
+                      has_one :author, class_name: "User", required: false
+                    end
+                  RUBY
+
+                  expected = indented(<<~RBI, 4)
+                    sig { returns(T.nilable(::User)) }
+                    def author; end
+
+                    sig { params(value: T.nilable(::User)).void }
+                    def author=(value); end
+                  RBI
+                  assert_includes(
+                    rbi_for(:Post, compiler_options: { ActiveRecordAssociationTypes: "persisted" }),
+                    expected,
+                  )
+                end
+
+                it "generates non-nilable types if :required is true" do
+                  add_ruby_file("schema.rb", <<~RUBY)
+                    ActiveRecord::Migration.suppress_messages do
+                      ActiveRecord::Schema.define do
+                        create_table :posts do |t|
+                        end
+                      end
+                    end
+                  RUBY
+
+                  add_ruby_file("user.rb", <<~RUBY)
+                    class User < ActiveRecord::Base
+                    end
+                  RUBY
+
+                  add_ruby_file("post.rb", <<~RUBY)
+                    class Post < ActiveRecord::Base
+                      has_one :author, class_name: "User", required: true
+                    end
+                  RUBY
+
+                  expected = indented(<<~RBI, 4)
+                    sig { returns(::User) }
+                    def author; end
+
+                    sig { params(value: ::User).void }
+                    def author=(value); end
+                  RBI
+                  assert_includes(
+                    rbi_for(:Post, compiler_options: { ActiveRecordAssociationTypes: "persisted" }),
+                    expected,
+                  )
+                end
+              end
+
+              describe "belongs to relations" do
+                it "falls back to the default if :optional is not an option" do
+                  add_ruby_file("schema.rb", <<~RUBY)
+                    ActiveRecord::Migration.suppress_messages do
+                      ActiveRecord::Schema.define do
+                        create_table :default_required_posts do |t|
+                          t.references(:author, null: true)
+                        end
+
+                        create_table :default_optional_posts do |t|
+                          t.references(:author, null: true)
+                        end
+                      end
+                    end
+                  RUBY
+
+                  add_ruby_file("user.rb", <<~RUBY)
+                    class User < ActiveRecord::Base
+                    end
+                  RUBY
+
+                  add_ruby_file("default_required_post.rb", <<~RUBY)
+                    class DefaultRequiredPost < ActiveRecord::Base
+                      def self.belongs_to_required_by_default
```

Comment:
It seems important to me to test that we respect the default configuration, although this feels like a weak way to test this behavior. I could not figure out how to set the Rails configuration option directly.

Reply:
I think you should be able to set `ActiveRecord::Base.belongs_to_required_by_default = true` in the `schema.rb` file to set this this as default.

This is what Active Record tests seems to be doing: https://github.com/rails/rails/blob/8b0c8079c6ee8460354f8b3cbe2296d3f3f3b21e/activerecord/test/cases/associations/required_test.rb#L27-L40